### PR TITLE
cheesy-api

### PR DESCRIPTION
**Pull request by no1621name**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

If adding a new definition:
- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x] If this is for an npm package, match the name. If not, do not conflict with the name of an npm package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [x] `tslint.json` [should contain](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#linter-tslintjson) `{ "extends": "@definitelytyped/dtslint/dt.json" }`, and no additional rules.
- [x] `tsconfig.json` [should have](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#tsconfigjson) `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

### CODE REVIEW

**Comment by no1621name** [comment]
help...

File: types/cheesy-api/.eslintrc.json [hunk 1]
```diff
@@ -0,0 +1,3 @@
+{
+	"rules": {}
```
Comment (DangerBotOSS) [inline types/cheesy-api/.eslintrc.json:2]: - :no_entry_sign: The root .editorconfig style specifies spaces for whitespace. Please use spaces in new or changed types.

File: types/cheesy-api/cheesy-api-tests.ts [hunk 2]
```diff
@@ -0,0 +1,4 @@
+const obj: Subject = {
+	name: 'Someone',
```
Comment (DangerBotOSS) [inline types/cheesy-api/cheesy-api-tests.ts:2]: - :no_entry_sign: The root .editorconfig style specifies spaces for whitespace. Please use spaces in new or changed types.

File: types/cheesy-api/package.json [hunk 3]
```diff
@@ -0,0 +1,4 @@
+{
+	"name": "cheesy-api-types",
```
Comment (DangerBotOSS) [inline types/cheesy-api/package.json:2]: - :no_entry_sign: The root .editorconfig style specifies spaces for whitespace. Please use spaces in new or changed types.

File: types/cheesy-api/tslint.json [hunk 4]
```diff
@@ -0,0 +1,6 @@
+{
+	"extends": "@definitelytyped/dtslint/dt.json",
```
Comment (DangerBotOSS) [inline types/cheesy-api/tslint.json:2]: - :no_entry_sign: The root .editorconfig style specifies spaces for whitespace. Please use spaces in new or changed types.

**Comment by no1621name** [comment]
finally i did it!

File: types/cheesy-api/.eslintrc.json (hunk 1 above)
Comment (peterblazejewicz) [reply types/cheesy-api/.eslintrc.json:2]: Please run `prettier` task on this directory:
https://github.com/DefinitelyTyped/DefinitelyTyped#common-mistakes

**Comment by peterblazejewicz** [comment]
@no1621name is this is a private repository/project?

**Comment by no1621name** [comment]
@peterblazejewicz I'm going to do some corrections and then I'll open it

**Comment by no1621name** [comment]
@peterblazejewicz, can you accept my PR please?

**Comment by peterblazejewicz** [comment]
> @peterblazejewicz, can you accept my PR please?

Is this a public code? DT usually does not host a code for private, unpu blished packages.

**Comment by no1621name** [comment]
@peterblazejewicz well, if i open code it will help. Right?

**Comment by peterblazejewicz** [comment]
if this is still closed project, you can prepare everything wihtin NPM package, so the declaraton fles ships with the package source when published:
https://www.typescriptlang.org/docs/handbook/declaration-files/publishing.html#including-declarations-in-your-npm-package
At the moment I don't think that is OK to expect code to be reviewed, when there is no access to source code or public documentation of the api

**Comment by no1621name** [comment]
ok, i understood you. thanks

File: types/cheesy-api/.eslintrc.json [hunk 5]
```diff
@@ -0,0 +1,3 @@
+{
```
Comment (sheetalkamat) [inline types/cheesy-api/.eslintrc.json:1]: Remove this file

File: types/cheesy-api/tslint.json [hunk 6]
```diff
@@ -0,0 +1,6 @@
...
+	"rules": {
```
Comment (sheetalkamat) [inline types/cheesy-api/tslint.json:3]: This should be removed and fix the error as suggestion

**Review by sheetalkamat — CHANGES_REQUESTED** [review]
There is already https://github.com/no1621name/cheesy-api/blob/master/index.d.ts here and i dont think this type should be needed on Definitely typed

**Pull request closed without merging**